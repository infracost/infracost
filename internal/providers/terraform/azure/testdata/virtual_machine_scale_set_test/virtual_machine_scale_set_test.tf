provider "azurerm" {
  skip_provider_registration = true
  features {}
}

resource "azurerm_resource_group" "example" {
  name     = "example-resources"
  location = "West Europe"
}

resource "azurerm_virtual_network" "example" {
  name                = "acctvn"
  address_space       = ["10.0.0.0/16"]
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
}

resource "azurerm_subnet" "example" {
  name                 = "acctsub"
  resource_group_name  = azurerm_resource_group.example.name
  virtual_network_name = azurerm_virtual_network.example.name
  address_prefixes     = ["10.0.2.0/24"]
}

resource "azurerm_virtual_machine_scale_set" "linux" {
  name                = "mytestscaleset-1"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name

  upgrade_policy_mode = "Manual"

  sku {
    name     = "Standard_F2"
    tier     = "Standard"
    capacity = 2
  }

  storage_profile_image_reference {
    publisher = "Canonical"
    offer     = "UbuntuServer"
    sku       = "16.04-LTS"
    version   = "latest"
  }

  storage_profile_os_disk {
    name              = ""
    caching           = "ReadWrite"
    create_option     = "FromImage"
    managed_disk_type = "Standard_LRS"
  }

  storage_profile_data_disk {
    lun               = 0
    caching           = "ReadWrite"
    create_option     = "Empty"
    managed_disk_type = "StandardSSD_LRS"
    disk_size_gb      = 32
  }

  storage_profile_data_disk {
    lun               = 1
    caching           = "ReadWrite"
    create_option     = "Empty"
    managed_disk_type = "Standard_LRS"
    disk_size_gb      = 10
  }

  os_profile {
    computer_name_prefix = "testvm"
    admin_username       = "myadmin"
  }

  os_profile_linux_config {
    disable_password_authentication = true
  }

  network_profile {
    name    = "terraformnetworkprofile"
    primary = true

    ip_configuration {
      name                                   = "TestIPConfiguration"
      primary                                = true
      subnet_id                              = azurerm_subnet.example.id
      load_balancer_backend_address_pool_ids = ["fake"]
      load_balancer_inbound_nat_rules_ids    = ["fake"]
    }
  }
}

resource "azurerm_virtual_machine_scale_set" "windows" {
  name                = "mytestscaleset-1"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name

  upgrade_policy_mode = "Manual"

  sku {
    name     = "Standard_F2"
    tier     = "Standard"
    capacity = 3
  }

  storage_profile_image_reference {
    publisher = "Canonical"
    offer     = "UbuntuServer"
    sku       = "16.04-LTS"
    version   = "latest"
  }

  storage_profile_os_disk {
    name              = ""
    caching           = "ReadWrite"
    create_option     = "FromImage"
    managed_disk_type = "Standard_LRS"
  }

  storage_profile_data_disk {
    lun               = 0
    caching           = "ReadWrite"
    create_option     = "Empty"
    managed_disk_type = "StandardSSD_LRS"
    disk_size_gb      = 32
  }

  storage_profile_data_disk {
    lun               = 1
    caching           = "ReadWrite"
    create_option     = "Empty"
    managed_disk_type = "Standard_LRS"
    disk_size_gb      = 10
  }

  os_profile {
    computer_name_prefix = "testvm"
    admin_username       = "myadmin"
  }

  os_profile_windows_config {
  }

  network_profile {
    name    = "terraformnetworkprofile"
    primary = true

    ip_configuration {
      name                                   = "TestIPConfiguration"
      primary                                = true
      subnet_id                              = azurerm_subnet.example.id
      load_balancer_backend_address_pool_ids = ["fake"]
      load_balancer_inbound_nat_rules_ids    = ["fake"]
    }
  }
}
