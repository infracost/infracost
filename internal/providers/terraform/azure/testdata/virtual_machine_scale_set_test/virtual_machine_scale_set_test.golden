
 Name                                                    Monthly Qty  Unit                      Monthly Cost 
                                                                                                             
 azurerm_virtual_machine_scale_set.linux                                                                     
 ├─ Instance usage (pay as you go, Standard_F2)                1,460  hours                          $166.44 
 ├─ storage_os_disk                                                                                          
 │  ├─ Storage (S4)                                                1  months                           $1.54 
 │  └─ Disk operations                            Monthly cost depends on usage: $0.0005 per 10k operations  
 ├─ storage_data_disk                                                                                        
 │  ├─ Storage (E4)                                                1  months                           $2.40 
 │  └─ Disk operations                            Monthly cost depends on usage: $0.002 per 10k operations   
 └─ storage_data_disk                                                                                        
    ├─ Storage (S4)                                                1  months                           $1.54 
    └─ Disk operations                            Monthly cost depends on usage: $0.0005 per 10k operations  
                                                                                                             
 azurerm_virtual_machine_scale_set.windows                                                                   
 ├─ Instance usage (hybrid benefit, Standard_F2)               2,190  hours                          $249.66 
 ├─ storage_os_disk                                                                                          
 │  ├─ Storage (S4)                                                1  months                           $1.54 
 │  └─ Disk operations                                           100  10k operations                   $0.05 
 ├─ storage_data_disk                                                                                        
 │  ├─ Storage (E4)                                                1  months                           $2.40 
 │  └─ Disk operations                                           200  10k operations                   $0.40 
 └─ storage_data_disk                                                                                        
    ├─ Storage (S4)                                                1  months                           $1.54 
    └─ Disk operations                                           200  10k operations                   $0.10 
                                                                                                             
 OVERALL TOTAL                                                                                       $427.59 
──────────────────────────────────
5 cloud resources were detected:
∙ 2 were estimated, all of which include usage-based costs, see https://infracost.io/usage-file
∙ 3 were free:
  ∙ 1 x azurerm_resource_group
  ∙ 1 x azurerm_subnet
  ∙ 1 x azurerm_virtual_network